api/search.go

```diff
@@ -241,10 +275,22 @@ func sanitiseAndValidateQuery(ctx context.Context, params url.Values) (string, e
 		log.Info(ctx, "rejecting query as it contains special characters", log.Data{"query": sanitisedQuery})
 		return "", errors.New("invalid characters in query")
 	}
-
 	return sanitisedQuery, nil
 }
 
+func parseCDID(ctx context.Context, params url.Values) (cdids []string, err error) {
+	cdidParam := paramGet(params, ParamCDIDs, "")
+	if cdidParam != "" {
+		cdids = strings.Split(cdidParam, ",")
+		disallowed, validationErr := validateCDIDs(cdids)
+		if validationErr != nil {
+			log.Warn(ctx, validationErr.Error(), log.Data{"param": ParamCDIDs, "value": cdidParam, "disallowed": disallowed})
```

Comment:
[3/3] this error message (2nd arg) is non-static, and we prefer static ones (e.g. searchability/filtering in kibana)
plus, the `disallowed` IDs are now logged twice